### PR TITLE
Prepare admission and CRD e2e tests to be promoted to conformance

### PR DESCRIPTION
**Pull request by jpbetz**
- [x] Remove all skips
- [x] Remove time.Sleep calls
- [x] Fix all e2e tests to use v1 instead of v1beta1 (this was done but got clobbered in the recent merges)

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Admission webhook and CRD e2e tests will be added to conformance as part of their promotion to GA. Conformance tests may not be skipped.

```release-note
NONE
```

@liggitt @roycaihw @sttts @caesarxuchao 

/sig api-machinery
/area admission
/area custom-resources
/area conformance
/priority important-soon

### CODE REVIEW

File: test/e2e/apimachinery/crd_conversion_webhook.go [hunk 1]
```diff
@@ -127,9 +127,6 @@ var _ = SIGDescribe("CustomResourceConversionWebhook", func() {
 		client = f.ClientSet
 		namespaceName = f.Namespace.Name
 
-		// Make sure the relevant provider supports conversion webhook
-		framework.SkipUnlessServerVersionGTE(serverCRDConversionWebhookVersion, f.ClientSet.Discovery())
```
Comment (liggitt) [inline test/e2e/apimachinery/crd_conversion_webhook.go]: drop `serverCRDConversionWebhookVersion`?

File: test/e2e/apimachinery/crd_publish_openapi.go [hunk 2]
```diff
@@ -54,10 +54,6 @@ var (
 var _ = SIGDescribe("CustomResourcePublishOpenAPI", func() {
 	f := framework.NewDefaultFramework("crd-publish-openapi")
 
-	ginkgo.BeforeEach(func() {
-		framework.SkipUnlessServerVersionGTE(crdPublishOpenAPIVersion, f.ClientSet.Discovery())
```
Comment (liggitt) [inline test/e2e/apimachinery/crd_publish_openapi.go]: drop `crdPublishOpenAPIVersion`?

File: test/e2e/apimachinery/crd_watch.go [hunk 3]
```diff
@@ -46,8 +46,6 @@ var _ = SIGDescribe("CustomResourceDefinition Watch", func() {
 		*/
 		ginkgo.It("watch on custom resource definition objects", func() {
 
-			framework.SkipUnlessServerVersionGTE(crdVersion, f.ClientSet.Discovery())
```
Comment (liggitt) [inline test/e2e/apimachinery/crd_watch.go]: drop `crdVersion`?

File: test/e2e/apimachinery/webhook.go [hunk 4]
```diff
@@ -91,15 +91,6 @@ var _ = SIGDescribe("AdmissionWebhook", func() {
 		client = f.ClientSet
 		namespaceName = f.Namespace.Name
 
-		// Make sure the relevant provider supports admission webhook
-		framework.SkipUnlessServerVersionGTE(serverWebhookVersion, f.ClientSet.Discovery())
```
Comment (liggitt) [inline test/e2e/apimachinery/webhook.go]: drop `serverWebhookVersion`?

**Comment by liggitt** [comment]
> * Update CustomResourceDefinitionVersion to of CRs in e2e tests to v1? This might confuse apisnoop?

that's probably not necessary... apisnoop should be able to filter out example.com-suffixed APIs

**Comment by jpbetz** [comment]
@liggitt The time.Sleep removal is passing now, and it really speeds things up, but it's fairly ugly. Any suggestions on how to clean it up welcome. I have one TODO in there about the "slow admit" testing, which is the one that's currently in the worst shape.

Skip removals, Sleep removals, and v1beta1->v1 cleanup are in separate commits.

**Comment by jpbetz** [comment]
rebased

**Comment by jpbetz** [comment]
@liggitt time.Sleep fixes are now passing, and much faster than the mandatory 10s waits.  There were a couple that I couldn't remove, and one that I can remove but left a TODO for, which i'll try to get to tomorrow. I'm not super happy with how the code fell out. Any suggestions welcome. Maybe inline the wait.poll calls instead of adding new functions?

Skip removal, sleep removal and v1beta1->v1 fixup are in separate commits

File: test/e2e/apimachinery/webhook.go [hunk 5]
```diff
@@ -1126,11 +1128,12 @@ func registerValidatingWebhookForWebhookConfigurations(f *framework.Framework, c
 				Rules: []admissionregistrationv1.RuleWithOperations{{
 					Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Delete},
 					Rule: admissionregistrationv1.Rule{
-						APIGroups:   []string{"admissionregistration.k8s.io"},
+						APIGroups:   []string{"", "admissionregistration.k8s.io"},
 						APIVersions: []string{"*"},
 						Resources: []string{
 							"validatingwebhookconfigurations",
 							"mutatingwebhookconfigurations",
+							"configmaps",
```
Comment (liggitt) [inline test/e2e/apimachinery/webhook.go]: put this in a separate rule, with a comment why we're including it?

File: test/e2e/apimachinery/webhook.go [hunk 6]
```diff
@@ -1126,11 +1128,12 @@ func registerValidatingWebhookForWebhookConfigurations(f *framework.Framework, c
 				Rules: []admissionregistrationv1.RuleWithOperations{{
 					Operations: []admissionregistrationv1.OperationType{admissionregistrationv1.Delete},
 					Rule: admissionregistrationv1.Rule{
-						APIGroups:   []string{"admissionregistration.k8s.io"},
+						APIGroups:   []string{"", "admissionregistration.k8s.io"},
```
Comment (liggitt) [inline test/e2e/apimachinery/webhook.go]: separate rule (apiGroup:"" doesn't make sense for validatingwebhookconfigurations/mutatingwebhookconfigurations)

File: test/e2e/apimachinery/webhook.go [hunk 7]
```diff
@@ -1947,8 +1960,15 @@ func registerSlowWebhook(f *framework.Framework, configName string, context *cer
...
 
-	// The webhook configuration is honored in 10s.
-	time.Sleep(10 * time.Second)
+	// TODO: Remove this and instead register the slow webhook only once and then updated it and wait
+	// for the transitions to occur, which are all possible to detect
+	if *policy == admissionregistrationv1.Fail && timeout != nil && *timeout <= 5 {
+		err = waitAlwaysAllowDelayWebhookReady(f, client)
+		framework.ExpectNoError(err, "waiting for always allow config map with delay webhook to be ready")
+	} else {
+		// Since there is no client visible change, wait 10s for the webhook configuration to be honored.
```
Comment (liggitt) [inline test/e2e/apimachinery/webhook.go]: for failurePolicy:Ignore, if a request completes faster than the slow webhook timeout, we know it's not effective yet, right?

**Comment by liggitt** [comment]
> they're verifying that a webhook for webhooks is a noop, which is undetectable (unless there is some trick I don't know)

would it make sense to always include an identical marker webhook in the configuration that lets us know when a configuration has been observed and is effective? a webhook that selected configmaps labeled "e2e-marker" and routed them to the denyall webhook would let us uniformly test when a config was effective (and all webhooks in the config become effective at the same time). that would let us use the same "waitForEffectiveWebhook" method for all tests (repeatedly submit a configmap labeled e2e-marker until it fails).

**Comment by jpbetz** [comment]
> would it make sense to always include an identical marker webhook in the configuration that lets us know when a configuration has been observed and is effective? a webhook that selected configmaps labeled "e2e-marker" and routed them to the denyall webhook would let us uniformly test when a config was effective (and all webhooks in the config become effective at the same time). that would let us use the same "waitForEffectiveWebhook" method for all tests (repeatedly submit a configmap labeled e2e-marker until it fails).

This is just what I need. Will update PR with this shortly.

**Comment by jpbetz** [comment]
@liggitt I've switched to using markers. Much simpler. I created a separate namespace for the markers to isolate both the markers and the webhooks from each other.

**Comment by jpbetz** [comment]
It would be possible to move both the marker webhook registration and the wait step all into `createValidatingWebhookConfiguration`.. Worth it?

**Comment by liggitt** [comment]
/lgtm
/approve

> It would be possible to move both the marker webhook registration and the wait step all into `create{Validating,Mutating}WebhookConfiguration`.. Worth it?

I think this is fine as-is

**Pull request merged**